src/backend/dx12/src/device.rs

```diff
@@ -1496,11 +1496,104 @@ impl d::Device<B> for Device {
-                &d3d12::IID_ID3D12PipelineState,
-                &mut pipeline as *mut *mut _ as *mut *mut _)
+        let hr = if desc.depth_stencil.unwrap_or(unsafe { mem::zeroed() }).depth_bounds {
+            // The DepthBoundsTestEnable option isn't available in the original D3D12_GRAPHICS_PIPELINE_STATE_DESC struct.
+            // Instead, we must use the newer subobject stream method.
+            match self.raw.cast::<d3d12::ID3D12Device2>() {
+                Err(hr) => hr,
+                Ok(device2) => {
+                    #[repr(C)]
+                    struct Subobject<T> {
+                        subobject_align: [usize; 0], // Subobjects must have the same alignment as pointers.
+                        subobject_type: d3d12::D3D12_PIPELINE_STATE_SUBOBJECT_TYPE,
+                        subobject: T,
+                    }
+                    impl<T> Subobject<T> {
+                        fn new(subobject_type: d3d12::D3D12_PIPELINE_STATE_SUBOBJECT_TYPE, subobject: T) -> Self {
+                            Subobject {
+                                subobject_align: [],
+                                subobject_type,
+                                subobject,
+                            }
+                        }
+                    }
+                    #[repr(C)]
+                    struct SubobjectStream {
+                        root_signature: Subobject<*mut d3d12::ID3D12RootSignature>,
+                        vs: Subobject<d3d12::D3D12_SHADER_BYTECODE>,
+                        ps: Subobject<d3d12::D3D12_SHADER_BYTECODE>,
+                        ds: Subobject<d3d12::D3D12_SHADER_BYTECODE>,
+                        hs: Subobject<d3d12::D3D12_SHADER_BYTECODE>,
+                        gs: Subobject<d3d12::D3D12_SHADER_BYTECODE>,
+                        stream_output: Subobject<d3d12::D3D12_STREAM_OUTPUT_DESC>,
+                        blend: Subobject<d3d12::D3D12_BLEND_DESC>,
+                        sample_mask: Subobject<UINT>,
+                        rasterizer: Subobject<d3d12::D3D12_RASTERIZER_DESC>,
+                        depth_stencil: Subobject<d3d12::D3D12_DEPTH_STENCIL_DESC1>,
+                        input_layout: Subobject<d3d12::D3D12_INPUT_LAYOUT_DESC>,
+                        ib_strip_cut_value: Subobject<d3d12::D3D12_INDEX_BUFFER_STRIP_CUT_VALUE>,
+                        primitive_topology: Subobject<d3d12::D3D12_PRIMITIVE_TOPOLOGY_TYPE>,
+                        render_target_formats: Subobject<d3d12::D3D12_RT_FORMAT_ARRAY>,
+                        depth_stencil_format: Subobject<dxgiformat::DXGI_FORMAT>,
+                        sample_desc: Subobject<dxgitype::DXGI_SAMPLE_DESC>,
+                        node_mask: Subobject<UINT>,
+                        cached_pso: Subobject<d3d12::D3D12_CACHED_PIPELINE_STATE>,
+                        flags: Subobject<d3d12::D3D12_PIPELINE_STATE_FLAGS>,
+                    }
+                    let mut pss_stream = SubobjectStream {
```

Comment:
this function body became quite heavy, with a lot of meat indented. Let's move the bulk of logic into a helper `create_pipeline_stream` method returning `SubobjectStream` (which also means moving the types out of the function body)